### PR TITLE
more friendly message for no-unescaped-entities

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

File: lib/rules/no-unescaped-entities.js [hunk 1]
```diff
@@ -59,10 +71,18 @@ module.exports = {
         for (let j = 0; j < entities.length; j++) {
           for (let index = 0; index < rawLine.length; index++) {
             const c = rawLine[index];
-            if (c === entities[j]) {
+            if (typeof entities[j] === 'string') {
```
Comment (ljharb) [inline lib/rules/no-unescaped-entities.js]: what is this line for? i don't see any new tests.

Comment (stevemao) [reply lib/rules/no-unescaped-entities.js]: The structure of `DEFAULTS` changed to an array of objects. I'll add some tests.

**Comment by stevemao** [comment]
BTW, why is `&` not in the default?

File: lib/rules/no-unescaped-entities.js (hunk 1 above)
Comment (stevemao) [reply lib/rules/no-unescaped-entities.js]: I have added a test which takes an option, `forbid`.

**Comment by stevemao** [comment]
@ljharb Do you also want to change the option to be `array` of

```js
{
  char,
  alternatives,
}
```

This would become a breaking change (change from `array` of `string`s)?

**Comment by ljharb** [comment]
I definitely want to avoid a breaking change.

File: tests/lib/rules/no-unescaped-entities.js [hunk 2]
```diff
@@ -186,7 +186,20 @@ ruleTester.run('no-unescaped-entities', rule, {
...
+        var Hello = createReactClass({
+          render: function() {
+            return <>foo & bar</>;
+          }
+        });
+      `,
+      parser: 'babel-eslint',
+      errors: [{message: 'HTML entities, \`&\` , must be escaped.'}],
+      options: [{
+        forbid: ['&']
```
Comment (ljharb) [inline tests/lib/rules/no-unescaped-entities.js]: Can we add a forbid test that use the default parser?

Comment (stevemao) [reply tests/lib/rules/no-unescaped-entities.js]: @ljharb done :)

File: docs/rules/no-unescaped-entities.md [hunk 3]
```diff
@@ -83,4 +83,14 @@ Overwrite the default forbidden entities array `['>', '"', '\'', '}']` with your
...
+"react/no-unescaped-entities": ["error", {"forbid": [{
+  char: ">",
+  alternatives: ['&gt;']
+}, {
+  char: "}",
+  alternatives: ['&#125;']
+}]}],
 ```
+
+Where `char` is a special character and `alternatives` is the correct escapes.
```
Comment (stevemao) [inline docs/rules/no-unescaped-entities.md]: Also added docs

Comment (ljharb) [reply docs/rules/no-unescaped-entities.md]: Why multiple alternatives?

File: tests/lib/rules/no-unescaped-entities.js (hunk 2 above)
Comment (ljharb) [reply tests/lib/rules/no-unescaped-entities.js]: maybe also tests that cover providing custom alternatives?

File: lib/rules/no-unescaped-entities.js [hunk 4]
```diff
@@ -28,10 +40,7 @@ module.exports = {
       type: 'object',
       properties: {
         forbid: {
-          type: 'array',
-          items: {
-            type: 'string'
```
Comment (ljharb) [inline lib/rules/no-unescaped-entities.js]: these should be tightly specified - either a string, or an object with the proper "char" + "alternatives" schema

Comment (stevemao) [reply lib/rules/no-unescaped-entities.js]: Yeah, I tried. I couldn't find a `Either` type with this lib.

**Comment by ljharb** [comment]
(I'm going to hold off merging it for awhile while the 7.12 release settles down)

**Pull request merged**